hyperspy/tests/drawing/test_plot_markers.py

```diff
@@ -472,4 +472,42 @@ def test_plot_eds_lines():
     s.plot(True)
     s.axes_manager.navigation_axes[0].index = 1
     return s._plot.signal_plot.figure
-    
+
+@update_close_figure
+def test_iterate_markers():
+    from skimage.feature import peak_local_max
+    import scipy.misc
+    ims = BaseSignal(scipy.misc.face()).as_signal2D([1,2])
+    index = np.array([peak_local_max(im.data, min_distance=100,
+                             num_peaks=4) for im in ims])
+    # Add multiple markers
+    for i in range(4):
+        xs = index[:, i, 1]
+        ys = index[:, i, 0]
+        m = markers.point(x=xs, y=ys, color='red')
+        ims.add_marker(m, plot_marker=True, permanent=True)
+        m = markers.text(x=10+xs, y=10+ys, text=str(i), color='k')
+        ims.add_marker(m, plot_marker=True, permanent=True)
+    xs = index[:, :, 1]
+    ys = index[:, :, 0]
+    m = markers.rectangle(np.min(xs, 1),
+                          np.min(ys, 1),
+                          np.max(xs, 1),
+                          np.max(ys, 1),
+                          color='green')
+    ims.add_marker(m, plot_marker=True, permanent=True)
+
+    for im in ims:
+        m_original = ims.metadata.Markers
+        m_iterated = im.metadata.Markers
+        for key in m_original.keys():
+            mo = m_original[key]
+            mi = m_iterated[key]
+            assert mo.__class__.__name__         == mi.__class__.__name__
+            assert mo.name                       == mi.name
+            assert mo.get_data_position('x1')    == mi.get_data_position('x1')
+            assert mo.get_data_position('y1')    == mi.get_data_position('y1')
+            assert mo.get_data_position('text')  == mi.get_data_position('text')
+            assert mo.marker_properties['color'] == \
+                                                   mi.marker_properties['color']
+    return ims
```

Comment:
and this?

Reply:
This test compares the markers from the sub-signal `ims` with the markers from the original signal `im`. It should check if the markers are replicated correctly when the signal is iterated into sub-signals; `for im in ims`.

Reply:
Ups! certainly not. I will remove that together with the update_close_figure.

Thanks for reviewing!